### DISCUSSION TITLE
Hybrid Public Talk Speaker with Media to Display

### DISCUSSION THREAD

**Discussion by Chipperchap** [sircharlo/meeting-media-manager#987] [Q&A] [answered]
Hi All,
Just wondering what procedure you are using for the scenario when a public talk speaker is giving his talk over ZOOM for an in-person audience as well as hybrid audience, and he also has media / pictures to display. 
Currently we spotlight the speaker so that he can be viewed on the K-Hall screens (ZOOM dual monitors + M<sup>3</sup> display off), but when a picture is displayed we need to turn M<sup>3</sup> on + "replace spotlight" with the Kingdom Hall host computer so that the speaker and ZOOM participants can see the image as well... then reverse the process and "replace spotlight" with the speaker again.
This method is hardly a difficulty, albeit a bit clunky.
Is that what you do or do you have another method?
I was speaking with a friend and he said "what about using picture in picture" so that the speaker can see the image using a separate "scene" in OBS... I have not thought about how to implement this and its possibilities, but before I investigate I thought I would ask if there was anything obvious that I was missing first via this very clever forum. 🤗

**Comment by mtdvlpr** [comment]
We (my congregation) are currently not supporting that scenario, since the guidelines (S-260) say the following:
> ... those giving public talks should be physically present at the Kingdom Hall. It is preferred that those who are assigned other parts also be physically present at the Kingdom Hall.

So when our public talk speaker is not able to be in the Kingdom Hall, we watch the public talk of the meeting available at JW stream.

For other parts that are allowed to be held over Zoom, we only show the picture for the in-person audience, because the extra steps necessary to also show the image for Zoom participants are too technical for some of our AV brothers.

I have heard from another brother who does use the picture in picture feature of OBS. It's just a matter of having another scene with both the camera and the media window as a source and then resizing them to your liking. You could even go as far as to add a filter to the camera source to try and hide the background and only show the speaker on top of the media window. I have no experience with it myself though.

**Reply by e-Feitosa** [reply]
I have sent a message to your e-mail address

**Comment by Chipperchap** [comment]
Hi @mtdvlpr - that is interesting. 
I find that from time to time there may be slight differences in instructions provided to various branches. In my branch territory we received instruction that allowed for "occasional" hybrid talks. (Ref: S-147-22.05-E par. 1)
Whereas our congregation cannot use JW.Stream for public talks because we have not been granted permission by the C.O. (Ref: as above).

> I have heard from another brother who does use the picture in picture feature of OBS. It's just a matter of having another scene with both the camera and the media window as a source and then resizing them to your liking. You could even go as far as to add a filter to the camera source to try and hide the background and only show the speaker on top of the media window. I have no experience with it myself though.

That is really interesting. I may look into how to do that. I am always interested in making such things as easy as possible for our A/V team who, like yours perhaps, find such things too technical.

Really appreciate your help and the reply. 😀

**Reply by e-Feitosa** [reply]
Hello, it is not so difficult to solve situation if you use the zoom enabled dual monitor option. I will try to explain:
Follow the numbers in the images
1. M3 with the pictures of the speech (speaker)
2. zoom in on the PC in the Kingdom Hall (KH)
3. Image of the speaker in the KH (When you disable the text in item 5)
4. When disable image on item 5, only speaker on KH is left
5. To display the image of the speaker, just click the PLAYER / STOP and then click the icon (5) to disable the image (text of the year).
It requires a little attention from the operator, but works well.

![image](https://user-images.githubusercontent.com/90984283/210158936-bc035be4-c8b8-45ef-810e-7669932aa432.png)

Olá, não é tão difícil de resolver situação se você usa o zoom ativado a opção monitor duplo. Vou tentar explicar:
Siga os números nas imagens
1. M3 com as imagens do discurso (orador)
2. Zoom no PC do Salão do Reino (KH)
3. Imagem do orador no KH (Quando desabilita o texto no item 5)
4. Quando desabilita a imagem no item 5, fica somente o orador no KH
5. Para apresentar a imagem do orador, basta clicar no PLAYER / STOP e depois clica no ícone (5) para desabilitar a imagem (texto do ano).
Exige um pouco de atenção do operador, mas funciona bem.

**Reply by Chipperchap** [reply]
Thank you for your interest and screenshots. Really appreciated.

With this method however, does it mean that the ZOOM participants do not see the images - only those "in-person" at the Hall?

**Reply by e-Feitosa** [reply]
Zoom participants will see the image, however, in zoom they will not see the image only if the speaker's screen is active the zoom function to highlight the speaker's screen.

**Reply by e-Feitosa** [reply]
> Thank you for your interest and screenshots. Really appreciated.
> 
> With this method however, does it mean that the ZOOM participants do not see the images - only those "in-person" at the Hall?

Another option is for KS to share the image using the zoom itself.

**Comment by mtdvlpr** [comment, marked as answer]
Hi @Chipperchap, I had outdated information. Turns out, we did get the same instructions you did. So I did some experimenting and I think I found a decent way of going about this:

1. In OBS, have a scene called "Zoom Speaker" or something that has as a source a window capture of the second Zoom screen (with dual monitors, the screen that's hidden under the media window on the second monitor in the KH)
2. In Zoom, **pin** the Zoom speaker to your second Zoom screen and **spotlight** yourself to everyone (the virtual camera of OBS)
3. In M³, select your "Zoom Speaker" scene and hide the media window

After doing this, you will have the following situation:
- You and the physical participants will be looking at the pinned speaker on the second monitor in the KH
- The Zoom participants (including the Zoom speaker himself) will be looking at the speaker via the second monitor in the KH

Now when the speaker wants to show an image (or video even), you can simply press play for the desired media and everyone will see the desired media right away (Zoom participants will now be looking at the configured media scene as usual). When the desired media should be hidden again, simply stop the media in M³ and hide the media window again.

When the Zoom speaker is done, simply select the desired OBS scene in M³ and show the media window again. Then (in the background), you can unpin the Zoom speaker (only necessary when there's another Zoom part which isn't likely for the weekend meeting) and remove the spotlight (so that Zoom participants will see other Zoom participants giving an answer) without anyone noticing.

The only possible downside to this, is the fact that the Zoom speaker is now looking at himself by default, but he can of course change that by switching to gallery view. Hope this helps!

**Reply by mtdvlpr** [reply]
An M³ feature that could go along with this workflow would be a toggle button in the presentation mode indicating that a zoom part is currently in progress which would automatically hide the media window for you after playing media. What do you think about this @sircharlo?

**Reply by sircharlo** [reply]
Nice idea! In that scenario, there would be two options: set up the Zoom scene as you described, or simply ask the Zoom attendant to spotlight the KH video feed when there are any pictures during a Zoom part. What do you think, would adding a toggle for in-person/Zoom add unnecessary visual clutter, or could it really be useful?

**Reply by mtdvlpr** [reply]
Depending on how often parts are done from Zoom and how many images such a part has, it could save you multiple manual clicks and provide for a smoother transition between media and speaker

**Reply by sircharlo** [reply]
Agreed. In that case, perhaps the "in-person part" toggle could show up only in the case where the Zoom scene is configured in the settings? We'd have to document the way this feature works, of course.

**Reply by mtdvlpr** [reply]
I like it! Maybe simplify it further. Besides the camera/media scenes, a zoom part scene selector in settings. If that scene is active, just always hide the media window after media is done.

**Reply by sircharlo** [reply]
Or maybe an on/off in-person toggle in the media controller. 
If off (not in person, Zoom part), then hide the OBS scene picker, set the scene to Zoom scene and always hide the media window after media is done.
If on (in person, not a Zoom part), then show the OBS scene picker. The OBS scene picker would not have the "media" or "Zoom" scenes in the list, to avoid confusion.

**Reply by mtdvlpr** [reply]
Love it!

**Reply by mtdvlpr** [reply]
Continues in #1020

**Comment by Chipperchap** [comment]
https://user-images.githubusercontent.com/35364657/210651202-7ba3c9d0-6b86-4a0e-b92d-edc0944e4927.mp4

(In this video I have created a PIP of what is showing on the dual screen just to assist with visualisation... of course the PIP would not exist in the actual setting at the K-Hall) 😁
